### PR TITLE
Remove RabbitMQ from erchef and omnibus cookbooks

### PR DESCRIPTION
**Pull request by stevendanna**
RabbitMQ is no longer shipped with Chef Server. RabbitMQ was
previously used for search indexing and the "actions" queue which was
consumed by Chef Analytics and Chef Automate 1.

While we did support external RabbitMQ for "actions" queue, both
products that consumed it are now End-of-Life.

# Breaking Changes

- The /controls endpoint now always returns HTTP 410 Gone. The
  /controls endpoint was used by Chef Client's audit mode. This
  feature was removed in Chef Client 15. In Chef Client 14, this
  change will produce a non-fatal error that results in the audit log
  being written to disk.

- The Chef Server will no longer send data to Chef Analytics. This
  product was declared end-of-life on December 31, 2018.

- The Chef Server will no longer send data to Chef Automate 1.x via
  the actions queue. This product was declared end-of-life on December
  31, 2019. The Chef Server still sends data to Chef Automate 2 and
  Chef Automate 1.x via the data-collector endpoint.

https://docs.chef.io/versions/#end-of-life-eol

Signed-off-by: Steven Danna <steve@chef.io>

### CODE REVIEW

**Review by PrajaktaPurohit — APPROVED** [review]

**Comment by PrajaktaPurohit** [comment]
Already tested that the adhoc builds are successful at: https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/1634

**Review by markan — APPROVED** [review]
Soooo much cleanup. Thank you! A few minor observations beyond the comments

* Might clean up the master README to remove rabbitmq reference
* The preflight_solr_validator.rb file might need cleanup.
* Have we checked that the dev-vm still works?
* I note that the rabbitmq service was removed in a prior PR; however some migrations still refer to it. See 018_reindex_migration among others. If we run start/stop services on a missing service what happens?

File: omnibus/files/private-chef-cookbooks/private-chef/recipes/config.rb [hunk 1]
```diff
@@ -86,36 +86,33 @@
...
     # more or less the same as they were before the change-over to
     # Elasticsearch.
-    node.default['private_chef']['rabbitmq']['enable'] = true
     node.default['private_chef']['opscode-expander']['enable'] = true
     node.default['private_chef']['opscode-solr4']['enable'] = false
     node.default['private_chef']['elasticsearch']['enable'] = false
 
-    node.default['private_chef']['opscode-erchef']['search_provider'] = 'solr' # solr, elasticsearch
-    node.default['private_chef']['opscode-erchef']['search_queue_mode'] = 'rabbitmq' # rabbitmq, batch, or inline
+    node.default['private_chef']['opscode-erchef']['search_provider'] = 'solr'
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply omnibus/files/private-chef-cookbooks/private-chef/recipes/config.rb:93]: We are going to hopefully handle some of this config in a follow up.

**Comment by stevendanna** [comment]
@markan Yup, those are all great points.  Our next PR will to be to fix up some more of the configuration and fix the validators.

Re the migrations, I need to do a bit more thinking about what to do there.  All of the migrations that still reference rabbit are from pre-12.0 Chef. We don't have great testing around upgrades from 11.

**Pull request merged**